core/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -102,19 +94,21 @@
 
   private synchronized void appendCurrentBuffer(byte[] buf, int offset, int length)
       throws IOException {
-    if (!mTachyonFS.requestSpace(length)) {
-      mCanWrite = false;
-
-      String msg =
-          "Local tachyon worker does not have enough " + "space (" + length + ") or no worker for "
-              + mFile.mFileId + " " + mBlockId;
-
-      throw new IOException(msg);
+    if (mAvailableBytes < length) {
+      long bytesRequested = mTachyonFS.requestSpace(mBlockId, length - mAvailableBytes);
```

Comment:
Can you explain why user should keep trying when the server says it can only write less than requested?  If I need to write 10 bytes but I can only write 1, that would normally imply I should try another nodnode and delete thisblock.